juriscraper/opinions/united_states/state/ala.py

```diff
@@ -0,0 +1,116 @@
+"""
+Scraper for Alabama Supreme Court
+CourtID: ala
+Court Short Name: Alabama
+Author: William Palin
+Court Contact:
+History:
+ - 2023-01-04: Created.
+Notes:
+ - This may be the only court (with the app. cts) that requires selenium
+"""
+
+import re
+
+from lxml.html import fromstring, tostring
+
+from juriscraper.OpinionSiteLinearWebDriven import OpinionSiteLinearWebDriven
+
+
+class Site(OpinionSiteLinearWebDriven):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.court_id = self.__module__
+        self.url = (
+            "https://judicial.alabama.gov/decision/supremecourtdecisions"
+        )
+        self.uses_selenium = True
+
+    def _process_html(self):
+        if not self.test_mode_enabled():
+            self.initiate_webdriven_session()
+
+        # Fetch from the last four Released documents
+        if not self.test_mode_enabled():
+            links = self.html.xpath('//*[@id="FileTable"]/tbody/tr/td/a')[:4]
+        else:
+            links = ["fake_link"]
+        for link in links:
+            if not self.test_mode_enabled():
+                # Fetch the PDF at the a_href link in the file table and parse it
+                self.webdriver.get(link.get("href"))
+                date_filed = link.text_content().split("day, ")[1].strip()
+                # Wait for pdf to render
+                self.webdriver.implicitly_wait(10000)
+                # And click next twice to make sure we fully render the PDF content
+                self.find_element_by_id("next").click()
+                self.find_element_by_id("next").click()
+
+                # Set the content for lxml parsing after this rendering.
+                self.html = fromstring(self.webdriver.page_source)
+            else:
+                super()._download()
+                date_filed = "JANUARY 4, 2023"
+
+            # Iterate over the elements to extract out the content
+            start = False
+            content = []
+            author = ""
+            for element in self.html.xpath("//*"):
+                if b"markedContent" in tostring(element):
```

Comment:
Let's capture this in a variable at the outset if we're going to use it a bunch of times. Notionally:

```
s = tostring(element)
```

Reply:
Yeah, sorry.  I dropped `tostring` completely- and I think you'll find it more palatable